Library/Homebrew/formula_installer.rb

```diff
@@ -55,6 +56,13 @@ def initialize(formula)
     @pour_failed   = false
   end
 
+  # called by install/reinstall/upgrade when no build tools are available
+  def self.check_build_flags
```

Comment:
It specifically checks for flags that would trigger compilation of a formula, rather than pouring. It's called when neither Xcode or the CLT is present, stopping the user before they actually get to formula installation (and are bombarded with unhelpful compilation failures).